api/src/main/java/io/opentelemetry/trace/attributes/SemanticAttributes.java

```diff
@@ -291,5 +291,47 @@
   public static final StringAttributeSetter THREAD_NAME =
       StringAttributeSetter.create("thread.name");
 
+  /** Type of the trigger on which the function is executed. */
+  public static final StringAttributeSetter FAAS_TRIGGER =
+      StringAttributeSetter.create("faas.trigger");
+  /** String containing the execution id of the function. */
+  public static final StringAttributeSetter FAAS_EXECUTION =
+      StringAttributeSetter.create("faas.execution");
+  /** Indicates that the serverless function is executed for the first time (aka cold start). */
+  public static final BooleanAttributeSetter FAAS_COLDSTART =
+      BooleanAttributeSetter.create("faas.coldstart");
+  /** The name of the invoked function. */
+  public static final StringAttributeSetter FAAS_INVOKED_NAME =
+      StringAttributeSetter.create("faas.invoked_name");
+  /** The cloud provider of the invoked function. */
+  public static final StringAttributeSetter FAAS_INVOKED_PROVIDER =
+      StringAttributeSetter.create("faas.invoked_provider");
+  /** The cloud region of the invoked function. */
+  public static final StringAttributeSetter FAAS_INVOKED_REGION =
+      StringAttributeSetter.create("faas.invoked_region");
+
+  // faas.trigger == datasource
```

Comment:
Should this comment be added to the Javadoc of the respective attributes somehow?

Reply:
Good idea, thanks